### PR TITLE
ENH: Implement Feret/Caliper distance metrics in Find Feature Sizes.

### LINKED ISSUE #944: ENH: Ability to determine maximum and minimum feret diameter of features.
### Is there an existing plan for this?

- [X] I have searched the existing discussions, release notes, and documentation.

### Description of the Feature, Filter, or Functionality?

This is a follow up to a previous post on the [Google help page](https://groups.google.com/g/dream3d-users/c/PBNWXduBT4I/m/NaRBwvuTCwAJ). Is it possible to implement a filter (or perhaps add this functionality to the ```Find Feature Shapes``` filter) to determine the feret diameter of features? Currently, we can compute the equivalent diameter and [Axis lengths/Euler angles of the "best-fit" ellipsoid](https://www.dream3d.io/nx_reference_manual/html/OrientationAnalysis/FindShapesFilter.html). This works well for ellipsoidal grains but not for other features, such as lack of fusion (LOF) pores in metals/alloys (first image below) or thin annealing twins (second image below). 

The ```Compute Feature Corners``` filter writes six values to the Features.csv file that users can then use to manually determine the XYZ bounding box of each feature, but this would not work to measure the feret diameter of the annealing twin in the second screenshot below, since these are oriented arbitrarily within models. Ideally, this functionality would report the maximum and minimum feret diameter of each feature.

image001
image

### Version

7.0.x (DREAM3DNX beta)

### What section did you foresee your suggestion falling in? [Further details may be required during triage process]

_No response_

### High Level Steps To Implement

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by imikejackson**
- https://github.com/BlueQuartzSoftware/simplnx/issues/944

- [ ] Unit test implemented for the added features
- [ ] Documentation updated to include the ordering of the data in the output array

### CODE REVIEW

File: src/simplnx/DataStructure/Geometry/RectGridGeom.cpp [hunk 1]
```diff
@@ -544,6 +543,24 @@ Point3D<float64> RectGridGeom::getPlaneCoords(usize idx) const
   return coords;
 }
 
+std::pair<Point3Df, Point3Df> RectGridGeom::getCellBounds(usize x, usize y, usize z) const
+{
+  auto xBnds = getXBounds();
+  auto yBnds = getYBounds();
+  auto zBnds = getZBounds();
```
Comment (JDuffeyBQ) [inline src/simplnx/DataStructure/Geometry/RectGridGeom.cpp]:
```suggestion
  const auto& xBnds = getXBoundsRef();
  const auto& yBnds = getYBoundsRef();
  const auto& zBnds = getZBoundsRef();
```

File: src/Plugins/SimplnxCore/src/SimplnxCore/Filters/CalculateFeatureSizesFilter.cpp [hunk 2]
```diff
@@ -291,6 +296,101 @@ Result<> CalculateFeatureSizesFilter::executeImpl(DataStructure& dataStructure,
...
+      if(shouldCancel)
+      {
+        return {};
+      }
+
+      for(uint32 y = 0; y < yDim; y++)
+      {
+        for(uint32 x = 0; x < xDim; x++)
+        {
+          index = (imageDims[1] * imageDims[0] * z) + (imageDims[0] * y) + x; // Index into featureIds array
```
Comment (JDuffeyBQ) [inline src/Plugins/SimplnxCore/src/SimplnxCore/Filters/CalculateFeatureSizesFilter.cpp:335]: Looks like this is only used inside the loop so the declaration can just become this line.

File: src/simplnx/DataStructure/Geometry/RectGridGeom.cpp [hunk 3]
```diff
@@ -544,6 +543,24 @@ Point3D<float64> RectGridGeom::getPlaneCoords(usize idx) const
...
+
+  Point3Df minCoords;
+  minCoords[0] = static_cast<float32>((*xBnds)[x]);
+  minCoords[1] = static_cast<float32>((*yBnds)[y]);
+  minCoords[2] = static_cast<float32>((*zBnds)[z]);
```
Comment (JDuffeyBQ) [inline src/simplnx/DataStructure/Geometry/RectGridGeom.cpp]:
```suggestion
  minCoords[0] = static_cast<float32>(xBnds[x]);
  minCoords[1] = static_cast<float32>(yBnds[y]);
  minCoords[2] = static_cast<float32>(zBnds[z]);
```

File: src/simplnx/DataStructure/Geometry/RectGridGeom.cpp [hunk 4]
```diff
@@ -544,6 +543,24 @@ Point3D<float64> RectGridGeom::getPlaneCoords(usize idx) const
...
+
+  Point3Df maxCoords;
+  maxCoords[0] = (*xBnds)[x + 1];
+  maxCoords[1] = (*yBnds)[y + 1];
+  maxCoords[2] = (*zBnds)[z + 1];
```
Comment (JDuffeyBQ) [inline src/simplnx/DataStructure/Geometry/RectGridGeom.cpp]:
```suggestion
  maxCoords[0] = xBnds[x + 1];
  maxCoords[1] = yBnds[y + 1];
  maxCoords[2] = zBnds[z + 1];
```

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]

**Comment by imikejackson** [comment]
Closing this for now as there isn't funding to work the issue.

**Pull request closed without merging**